wandb/sdk/interface/router.py

```diff
@@ -44,8 +44,10 @@ class MessageRouter:
     _pending_reqs: Dict[str, MessageFutureObject]
     _request_queue: "Queue[pb.Record]"
     _response_queue: "Queue[pb.Result]"
+    _mailbox: Optional[mailbox.Mailbox]
 
-    def __init__(self) -> None:
+    def __init__(self, mailbox: Optional[mailbox.Mailbox] = None) -> None:
```

Comment:
Why make it optional? (here and elsewhere)

Reply:
it is optional when using the router_relay class.  This is because routing from the socket in the server to the internal_process queues (see streams.py) does not use another mailbox.   Mailbox is only for the user process.